src/convertToObject.js

```diff
@@ -7,6 +7,29 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  const result = {};
+  const regex = /\n+|\t+/g;
```

Comment:
[CODE KNOWLEDGE] - The regex pattern you've used will replace newlines and tabs with a space, which is good for cleaning up the input. However, it might be more efficient to use string methods to trim whitespace at the beginning and end of the string instead of replacing all occurrences within the string. Consider using `trim()` or `trimStart()` and `trimEnd()` if needed.